src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1101,6 +1113,124 @@ SequenceDDLCommandsForTable(Oid relationId)
 }
 
 
+/*
+ * SequenceDependencyCommandList generates commands to record the dependency
+ * of sequences on tables on the worker. This dependency does not exist by
+ * default since the sequences and table are created separately, but it is
+ * necessary to ensure that the sequence is dropped when the table is
+ * dropped.
+ *
+ * This function is derived from getOwnedSequences_internal.
+ */
+static List *
+SequenceDependencyCommandList(Oid relationId)
+{
+	List *sequenceCommandList = NIL;
+	char *relationName = generate_qualified_relation_name(relationId);
```

Comment:
Can't we directly use `getOwnedSequences` here ?

Reply:
we need the pg_depend record to get the column index (refobjsubid)

Reply:
If we need column index too, I wonder if `ExtractColumnsOwningSequences` does the same thing then ?

Reply:
Hmz, that function returns column names instead of indexes, but thinking about this more, it would be good to actually use column names instead of column indexes, since there might be scenarios in which the indexes don't match (drop a column that precedes the sequence, rebalance). Will change that.